plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -7274,9 +7274,7 @@ public int compare(final UnmanagedInstanceTO.Disk disk1, final UnmanagedInstance
                 }
 
                 int extractInt(UnmanagedInstanceTO.Disk disk) {
-                    String num = disk.getLabel().replaceAll("\\D", "");
-                    // return 0 if no digits found
-                    return num.isEmpty() ? 0 : Integer.parseInt(num);
+                    return disk.getPosition() + (disk.getControllerUnit() * 100);
```

Comment:
@DK101010 In the code above this section we set ControllerUnit only for `VirtualIDEController` and `VirtualSCSIController` device type otherwise it remains null. Should we add a null check here?

Reply:
@shwstppr Good point, But how realistic is it to say that we have a volume with a position and without a controller id.? I talked with my administration and network colleagues and other controller types like NVME and SATA have also a controller id.
Therefore I think about it to add an exception in the else condition? What do you think?

Reply:
@DK101010 yes, exception seems appropriate

Reply:
@shwstppr I will add an Exception.

Reply:
@shwstppr @DK101010 will throwing an exception cause VM to be not ingested?

Reply:
@rhtyd damn, your are right, that was to quick from my side, the catch block catch all exceptions :(. 

Hmm ... is it ok to remove the catch block ? 

In my opinion the ingest process should be canceled if not all volumes ingestible. What do you think @rhtyd @shwstppr ?

Reply:
Yes, as long as that's acceptable

Reply:
I think IDE and generic SCSI (VirtualSCSIController) will cover all disk controllers. If we are getting any other type of VirtualController as a disk, an exception seems correct to me. However, the error message needs to be a bit more generic there as the same method is used for listing unmanaged isntances

Reply:
@shwstppr Sure, we use also only IDE, SCSI controller but for me it sounds critical in case of other/new controller to ingest vm's without volumes . I think from user perspektive it is inconsistent to see that the vm is ingested but is unusable.